### ISSUE TITLE
`for async` loops

### ISSUE THREAD

**Issue by rictic** [angular/clang-format#44]
New syntax for consuming async iterators! It looks like clang-format wants to create a block after `for` and before `await`.

input:

```typescript
async function f() {
  for await (const val of makeAsyncIter()) {
  }
}
```

output:

```typescript
async function f() {
  for
    await(const val of makeAsyncIter()) {}
}
```

I wasn't sure where the best place to file this is, so I also filed this bug at https://issuetracker.google.com/38267829 feel free to close either.

**Comment by mprobst** [comment]
Let's use the inner one to track.

**Comment by mprobst** [comment]
This has been released in 1.0.51.
